space-concordia-robotics/robotics-prototype
# Rostwo migration 594

# Assignee Section

## Description
The mcu_control package should be migrated entirely over to ROS2.

The following nodes were migrated over:
- CommsNode
- JoyCommsControl
- Anything else relevent for current rover and arm control
-The following third party related applications must be confirm as working as well:

- ~cv_camera_node for rover camera streams~
- AR tag detection with at least aruco tags working
- ROS2 wrappers for 3D Lidar and ZED2 camera

Then the README and environment setup must be migrated over:
- Update ROS install script to install ROS 2 instead
- Update systemd service to startup ROS 2 rover launch file

### Steps for Testing
Follow setup instructions in README. Then confirm the following works:
- Successful on board test of updated CommsNode for operating Wheels and Arm with dualshock 4 controller
- Camera streams from rover
- 3D Lidar and ZED2 camera working as expected on board
- Aruco AR tag detection on board
- ROS2 rover startup sequence working reilability

closes #594 

The approval from all software team leads is necessary before merging.

# Reviewer Section
Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
